pallets/cash/src/internal/events.rs

```diff
@@ -84,18 +86,18 @@ pub fn risk_adjusted_value<T: Config>(
     }
 }
 
+static TRACK_CHAIN_EVENTS_LOCK: AtomicU8 = AtomicU8::new(0);
+
 /// Incrementally perform the next step of tracking events from all the underlying chains.
 pub fn track_chain_events<T: Config>() -> Result<(), Reason> {
-    let mut lock = StorageLock::<Time>::new(b"cash::track_chain_events");
-    let result = match lock.try_lock() {
-        Ok(_guard) => {
-            // Note: chains could be parallelized
-            track_chain_events_on::<T>(ChainId::Eth)
-        }
-
-        _ => Err(Reason::WorkerBusy),
-    };
-    result
+    if TRACK_CHAIN_EVENTS_LOCK.fetch_add(1, Ordering::SeqCst) == 0 {
```

Comment:
Curious if this provides _different_ guarantees than the previous lock, regardless of the timeout issue? Does the `StorageLock` use the same underlying mechanism?

Reply:
Yeah I guess this only worked for native since the memory isn't shared by WASM threads, reverted to 120s deadline with storage lock